app/src/main/java/com/github/mateo762/myapplication/followers/UserRepositoryImpl.kt

```diff
@@ -0,0 +1,169 @@
+package com.github.mateo762.myapplication.followers
+
+import android.util.Log
+import com.github.mateo762.myapplication.models.UserEntity
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.*
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.tasks.await
+import javax.inject.Inject
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+class UserRepositoryImpl @Inject constructor(private val db: DatabaseReference, private val auth: FirebaseAuth) : UserRepository {
```

Comment:
Nice that you implemented this, but isn't it adding code to tested?

Reply:
We still don't have a way to test firebase implementation, it will exist somewhere, either in activity or in a separate file. It is more convenient to have them in a separate file, as other things can be tested easier with this approach

Reply:
Understandable. We needed this part but the testing is still importnat.